Updated llvm to version 6.0.1. The previous 6.0.0 had an incorrectly …

neilflood opened the pull request:
…declared symbol, discussed at https://reviews.llvm.org/D44140, which, amongst other things, broke py-numba. This version works fine with py-numba.

healther commented:
can you add the known conflicts directly. This makes `spack` fail at the concretisation stage, rather than letting the user wait until the build fails (or worse the execution of the program)

healther commented:
Personally I'm against removing versions from spack (reproducibility being a major reason for using spack in the first place). That being said versions were removed in the past, mostly due to unreasonable effort to keep legacy build system support around. But this is a decision that @adamjstewart or @tgamblin should comment on.

I'm happy if you just add all the conflicts that you know (in this case I guess only `py-numba`) and add a comment to the `6.0.0` version of LLVM explaining the shortcoming, preferably with a link to the llvm-issue.

neilflood commented:
OK, that makes perfect sense. Just added the relevant conflicts() to py-numba, added to this PR. My earlier comment about other things being affected was the result of googling the error message, and finding that other people had complained about it. These were outside the context of Spack, though.
